_api_app/app/Sites/Settings/SiteSettingsDataService.php

```diff
@@ -440,6 +443,73 @@ public function saveValueByPath($path, $value)
         return $ret;
     }
 
+    public function uploadFileByPath($data)
+    {
+        $path = $data['path'];
+        $file = $data['value'];
+
+        if (!$file->isValid()) {
+            return [
+                'error' => 'Upload failed.',
+                'status' => 500
+            ];
+        }
+
+        $validator = Validator::make($data, [
```

Comment:
If this validator validates the request, it should happen in the controller, since the controller works with requests. If it validates some kind of data in this method (it seems to be validating only the value field), then again there's no need for the `$data` array.